macosx/Torrent.mm

```diff
@@ -198,49 +198,63 @@ - (instancetype)initWithMagnetAddress:(NSString*)address location:(NSString*)loc
 
 - (instancetype)initWithHistory:(NSDictionary*)history lib:(tr_session*)lib forcePause:(BOOL)pause
 {
-    self = [self initWithPath:history[@"InternalTorrentPath"] hash:history[@"TorrentHash"] torrentStruct:NULL magnetAddress:nil
-                            lib:lib
-                     groupValue:history[@"GroupValue"]
-        removeWhenFinishSeeding:history[@"RemoveWhenFinishSeeding"]
-                 downloadFolder:history[@"DownloadFolder"] //upgrading from versions < 1.80
-         legacyIncompleteFolder:[history[@"UseIncompleteFolder"] boolValue] //upgrading from versions < 1.80
-             ?
-             history[@"IncompleteFolder"] :
-             nil];
-
-    if (self)
+    bool isMagnet = false;
```

Comment:
Please use objc types in objc code, otherwise it becomes hard to reason about
```suggestion
    BOOL isMagnet = NO;
```

Reply:
ive updated. staring at moreC++ than objc was the problem here....
:)

Reply:
ive updated. staring at more C++ than objc was the problem here....
:)

Reply:
ive updated. staring at more C++ than objc was the problem here....
:)